app/src/main/java/com/infomaniak/drive/ui/SaveExternalFilesActivity.kt

```diff
@@ -278,9 +297,45 @@ class SaveExternalFilesActivity : BaseActivity() {
         }
     }
 
-    private fun store(uri: Uri, name: String? = null, userId: Int, driveId: Int, folderId: Int): Boolean {
-        val folder = java.io.File(cacheDir, SHARED_FILE_FOLDER).apply { if (!exists()) mkdirs() }
+    private fun storeText(userId: Int, driveId: Int, folderId: Int): Boolean {
+        intent.getStringExtra(Intent.EXTRA_TEXT)?.let { text ->
+            val extension = if (text.isValidUrl()) ".url" else ".txt"
```

Comment:
The user should be shown the file name with its extension directly in fileNameEdit. And not impose it. like on iOS